src/store/Workspaces/index.ts

```diff
@@ -215,43 +226,62 @@ function unSubscribeToEnvironmentOutput(workspaceId: string): void {
   if (!callback) {
     return;
   }
-  WorkspaceClient.jsonRpcMasterApi.unSubscribeEnvironmentOutput(workspaceId, callback);
+  cheWorkspaceClient.jsonRpcMasterApi.unSubscribeEnvironmentOutput(workspaceId, callback);
   subscribedEnvironmentOutputCallbacks.delete(workspaceId);
 }
 
 // ACTION CREATORS - These are functions exposed to UI components that will trigger a state transition.
 // They don't directly mutate state, but they can have external side-effects (such as loading data).
 export const actionCreators: ActionCreators = {
 
+  updateDevWorkspaceStatus: (workspace: che.Workspace, message: IStatusUpdate): AppThunk<KnownAction, Promise<void>> => async (dispatch): Promise<void> => {
+    onStatusUpdateReceived(workspace, dispatch, message);
+  },
+
   requestWorkspaces: (): AppThunk<KnownAction, Promise<void>> => async (dispatch): Promise<void> => {
     dispatch({ type: 'REQUEST_WORKSPACES' });
 
     try {
-      const workspaces = await WorkspaceClient.restApiClient.getAll<che.Workspace>();
+      const workspaces = await cheWorkspaceClient.restApiClient.getAll<che.Workspace>();
+      const defaultNamespace = await cheWorkspaceClient.getDefaultNamespace();
+      const isDevWorkspaceEnabled = await devWorkspaceClient.isEnabled();
+      let allWorkspaces = workspaces;
+      if (isDevWorkspaceEnabled) {
+        const devworkspaces = await devWorkspaceClient.getAllWorkspaces(defaultNamespace);
+        allWorkspaces = allWorkspaces.concat(devworkspaces);
+      }
 
       // Unsubscribe
       subscribedWorkspaceStatusCallbacks.forEach((workspaceStatusCallback: WorkspaceStatusMessageHandler, workspaceId: string) => {
         unSubscribeToStatusChange(workspaceId);
       });
 
-      // Subscribe
-      workspaces.forEach(workspace => {
-        subscribeToStatusChange(workspace.id, dispatch);
+      // Only subscribe to v1 workspaces
+      workspaces.forEach((workspace: any) => {
```

Comment:
These are v1 workspaces only. Could we specify them?
(workspace: any) -> (workspace: che.Workspace)